optuna/integration/_lightgbm_tuner/optimize.py

```diff
@@ -272,10 +272,10 @@ def _postprocess(
             self.pbar.set_description(self.pbar_fmt.format(self.step_name, self.best_score))
             self.pbar.update(1)
 
-        trial.set_system_attr(_ELAPSED_SECS_KEY, elapsed_secs)
-        trial.set_system_attr(_AVERAGE_ITERATION_TIME_KEY, average_iteration_time)
-        trial.set_system_attr(_STEP_NAME_KEY, self.step_name)
-        trial.set_system_attr(_LGBM_PARAMS_KEY, json.dumps(self.lgbm_params))
+        trial._set_system_attr(_ELAPSED_SECS_KEY, elapsed_secs)
+        trial._set_system_attr(_AVERAGE_ITERATION_TIME_KEY, average_iteration_time)
+        trial._set_system_attr(_STEP_NAME_KEY, self.step_name)
+        trial._set_system_attr(_LGBM_PARAMS_KEY, json.dumps(self.lgbm_params))
```

Comment:
Could you call `storage.set_trial_system_attrs` instead?

Reply:
I'd say we can get the `storage` object via `trial._storage`. Although it's a private attribute, it's acceptable to use inside the optuna's implementation.

Reply:
Thank you for you advice! (and sorry for late reply)

I update my code to use `trial._storage.set_trial_system_attrs(trial._trial_id, key, value)` instead of `trial.set_system_attrs(key, value)`.

I also use `study._storage.set_study_system_attrs(study._study_id, key, value)` instead of `study.set_system_attrs(key, value)`.